hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/HoodieFlinkWriteClient.java

```diff
@@ -369,7 +372,14 @@ public void completeCompaction(
   protected List<WriteStatus> compact(String compactionInstantTime, boolean shouldComplete) {
     // only used for metadata table, the compaction happens in single thread
     try {
-      List<WriteStatus> writeStatuses = FlinkCompactHelpers.compact(compactionInstantTime, this);
+      HoodieFlinkCopyOnWriteTable flinkCopyOnWriteTable = new HoodieFlinkCopyOnWriteTable(
+          getConfig(), getEngineContext(), getHoodieTable().getMetaClient());
+      HoodieWriteMetadata<HoodieData<WriteStatus>> writeMetadata =
+          CompactHelpers.newInstance().compact(context, getHoodieTable(), flinkCopyOnWriteTable,
+              config, compactionInstantTime, this, new FlinkTaskContextSupplier(null));
```

Comment:
why create a new FlinkTaskContextSupplier? dont' we have one already locally ?

Reply:
A new FlinkTaskContextSupplier instance is created in the original code.  Since the table already has it, I changed it to reuse the existing FlinkTaskContextSupplier.